common/state_of_charge.hpp

```diff
@@ -0,0 +1,57 @@
+#include "devices/sensors/battery/max17043.hpp"
+#include "peripherals/hardware_counter.hpp"
+#include "utility/log.hpp"
+namespace sjsu::common
+{
+
+  class StateOfCharge
+  {
+    public:
+    sjsu::Max17043 &module_;
+    StateOfCharge(sjsu::Max17043 &module) : module_(module) {}
+
+    void Initialize()
+    {
+      module_.ModuleInitialize();
+    }
+
+    double BatteryPercentage()
+    {
+      units::voltage::volt_t battery_voltage = module_.GetVoltage();
+      double voltage_percent = double(battery_voltage);
```

Comment:
```suggestion
      double voltage_percent = static_cast<double>(battery_voltage);
```
I think this is the safer c++ way